Removing artisinal HTTP requests in favor of a Gofer client

abloom opened the pull request.

hgschmie commented on Makefile:
```diff
@@ -1,7 +1,7 @@
 HOST=localhost
 PORT=3000
-SINGULARITY_HOST=localhost
+SINGULARITY_HOST=vagrant-singularity
```
Keep localhost for the example.

hgschmie commented on queequeg.js:
```diff
@@ -1,26 +1,40 @@
+
+var port = process.env.PORT || 3000,
+    host = process.env.HOST || 'localhost',
+    hookPrefix = "/webook",
```
webhook ?

abloom replied:
This is the path you prefixed all of the callback urls with. I moved it up here because we need it to setup the routes for the express app and to pass into the client to build the callbacks.

abloom replied:
oh, this was a spelling issue. caught it!

hgschmie commented on queequeg.js:
```diff
@@ -1,26 +1,40 @@
+
+var singularity = new Singularity({
+    singularity: {
+      baseUrl: "http://" + singularity_host + ":" + singularity_port + singularity_base
```
the util.format style keeps the actual shape of the URL and inserts the values which is more readable.

hgschmie commented on queequeg.js:
```diff
@@ -32,113 +46,41 @@ function log_callback(type, request, response) {
-        response.on('end', function(chunk) {
-            post_response = post_response + (chunk || '');
+function addWebhook(hook, callback) {
+    console.log("== adding Webhook %s for request type %s", hookBaseUrl + "/" + hook, hook);
```
mixes %s with string concat

hgschmie commented on queequeg.js:
```diff
@@ -32,113 +46,41 @@ function log_callback(type, request, response) {
 process.on('SIGINT', function() {
-    console.log("");
-    delete_webhooks(hooks,  function() { process.exit() });
+    console.log("CLEANUP");
```
the reason for the newline was to have this message not show up in the middle of another.

abloom replied:
I'm not sure this is guaranteed to work the way you expect. If other callbacks are in flight they could still intersperse with any messages generated here. The only way to guarantee it's not interrupted is to gather all the messages together and log the as one statement.

I'm removing the text because it's unlikely that this will get mixed up with other messages.

abloom commented:
Addressed all the feedback.

Made one more improvement - let express parse the incoming POST json.

Proof it works locally with a vagrant singularity image: https://gist.github.com/abloom/7acae957e82a63759ea8

hgschmie commented on Makefile:
```diff
@@ -1,9 +1,9 @@
-SINGULARITY_HOST=localhost
-SINGULARITY_PORT=7099
-SINGULARITY_BASE=/singularity
+HOST?=localhost
```
oh, you can have a ? there? What does it do?

abloom replied:
Conditionally set the variable. This way you can override them from the command line:

````bash
SINGULARITY_HOST=vagrant-singularity make server
````
